Makefile

```diff
@@ -254,6 +297,22 @@ ifeq ($(ARCH),amd64)
 	docker tag calico/felix:latest-$(ARCH) calico/felix:latest
 endif
 
+push-all: image-all imagetag $(addprefix sub-push-,$(ARCHES))
+sub-push-%:
+	$(MAKE) push ARCH=$* IMAGETAG=$(IMAGETAG)
+
+push: image imagetag
+	docker tag calico/felix:latest-$(ARCH) calico/felix:$(IMAGETAG)-$(ARCH)
```

Comment:
Again, think I'd prefer push-only with failure if the image isn't there.  Better to fail fast than silently do the wrong thing.

Reply:
Yeah @fasaxc and @caseydavenport convinced me of it a few minutes ago. I tried to address it in the latest commit to the [cni-plugin PR](https://github.com/projectcalico/cni-plugin/pull/516) by having the `push` target *only* push and a `tagimages` target tag correctly. It still simplifies greatly, but ensures that it only pushes the images if they exist.

Reply:
I will apply that here too (I accidentally hit "Comment" before I was done).